public/app/features/alerting/components/AlertingQueryRows.tsx

```diff
@@ -148,7 +149,7 @@ export class AlertingQueryRows extends PureComponent<Props, State> {
                       query={query.model}
                       onChange={(query) => this.onChangeQuery(query, index)}
                       timeRange={
-                        !isExpressionQuery(query.model)
+                        !isExpressionQuery(query.model) && query.relativeTimeRange
```

Comment:
Maybe don't use these anonymous functions but create proper class functions instead?

Reply:
hmm, we need to get the `index` variable from this scope so I think this is the most readable way of doing it. Right?

Reply:
Sure, I agree